modular_skyrat/modules/bonecrusher/bone_crusher.dm

```diff
@@ -0,0 +1,193 @@
+	var/modifiers = params2list(clickparams)
+	if(!wielded)
+		return
+	if(!proximity_flag && charged)//Mark a target, or mine a tile.
+		var/turf/proj_turf = user.loc
+		if(!isturf(proj_turf))
+			return
+		var/obj/projectile/destabilizer/D = new /obj/projectile/destabilizer(proj_turf)
+		for(var/t in trophies)
+			var/obj/item/crusher_trophy/T = t
+			T.on_projectile_fire(D, user)
+		D.preparePixelProjectile(target, user, modifiers)
+		D.firer = user
+		D.hammer_synced = src
+		playsound(user, 'sound/weapons/plasma_cutter.ogg', 100, TRUE)
+		D.fire()
+		charged = FALSE
+		update_appearance()
+		addtimer(CALLBACK(src, .proc/Recharge), charge_time)
+		return
+	if(proximity_flag && isliving(target))
+		var/mob/living/L = target
+		var/datum/status_effect/crusher_mark/CM = L.has_status_effect(STATUS_EFFECT_CRUSHERMARK)
+		if(!CM || CM.hammer_synced != src || !L.remove_status_effect(STATUS_EFFECT_CRUSHERMARK))
+			return
+		var/datum/status_effect/crusher_damage/C = L.has_status_effect(STATUS_EFFECT_CRUSHERDAMAGETRACKING)
+		if(!C)
+			C = L.apply_status_effect(STATUS_EFFECT_CRUSHERDAMAGETRACKING)
+		var/target_health = L.health
+		for(var/t in trophies)
+			var/obj/item/crusher_trophy/T = t
+			T.on_mark_detonation(target, user)
+		if(!QDELETED(L))
+			if(!QDELETED(C))
+				C.total_damage += target_health - L.health //we did some damage, but let's not assume how much we did
+			new /obj/effect/temp_visual/kinetic_blast(get_turf(L))
+			var/backstab_dir = get_dir(user, L)
+			var/def_check = L.getarmor(type = BOMB)
+			if((user.dir & backstab_dir) && (L.dir & backstab_dir))
+				if(!QDELETED(C))
+					C.total_damage += detonation_damage + backstab_bonus //cheat a little and add the total before killing it, so certain mobs don't have much lower chances of giving an item
+				L.apply_damage(detonation_damage + backstab_bonus, BRUTE, blocked = def_check)
+				playsound(user, 'sound/weapons/kenetic_accel.ogg', 100, TRUE) //Seriously who spelled it wrong
+			else
+				if(!QDELETED(C))
+					C.total_damage += detonation_damage
+				L.apply_damage(detonation_damage, BRUTE, blocked = def_check)
+
+/obj/item/bone_crusher/proc/Recharge()
+	if(!charged)
+		charged = TRUE
+		update_appearance()
+		playsound(src.loc, 'sound/weapons/kenetic_reload.ogg', 60, TRUE)
+
+/obj/item/bone_crusher/ui_action_click(mob/user, actiontype)
+	set_light_on(!light_on)
+	playsound(user, 'sound/weapons/empty.ogg', 100, TRUE)
+	update_appearance()
+
+
+/obj/item/bone_crusher/update_icon_state()
+	inhand_icon_state = "crusher[wielded]" // this is not icon_state and not supported by 2hcomponent
+	return ..()
+
+/obj/item/bone_crusher/update_overlays()
+	. = ..()
+	if(!charged)
+		. += "[icon_state]_uncharged"
+	if(light_on)
+		. += "[icon_state]_lit"
+
+
+///CRAFTING RECIPE
+
+/datum/crafting_recipe/bone_crusher
+	name = "Makeshift Proto-Kinetic Axe"
+	result = /obj/item/bone_crusher
+	reqs = list(/obj/item/fireaxe/boneaxe = 1,
+				/obj/item/gun/energy/kinetic_accelerator = 1,
```

Comment:
How are they meant to access a PKA?

Reply:
Legion corpses rarely drop them, killing the BDM drops one or two, trading with mining.. etc.